aeneas/src/x86/X86MacroAssembler.v3

```diff
@@ -320,6 +409,204 @@ class X86MacroAssembler extends X86Assembler {
 		}
 		if (frame != null) mach.runtime.src.recordFrameEnd(codeOffset());
 	}
+
+	def cast_f2ui(dest: X86Reg, a: SSERm) {
+		var sse_scratch = X86MachRegs.SSE_SCRATCH;
+		var save = esp.plusSSE(0x0);
+		sub.rm_i(esp, 0x8);
+		if (SSEAddr.?(a)) {
+			fld_d(SSEAddr.!(a));
+		} else {
+			movss_sm_s(save, SSEReg.!(a));
+			fld_d(save);
+		}
+		fisttp_q(save);
+		movd_r_rm(dest, save.toX86Addr());
+		add.rm_i(esp, 0x8);
+	}
+	def cast_d2ui(dest: X86Reg, a: SSERm) {
+		// TODO: convert double to unsigned int
+	}
+	def f2ui_fixup() {
```

Comment:
I don't quite understand this routine. For float/double convert to u32, you should be able to do a convert to i64 and then compare that result (unsigned 64-bit >=) with u32.max, no?

Reply:
That's essentially what it's doing. Basically the function is:
```
def f2ui(float f) -> u32 {
  if( f <= 0) return 0;
  if (f >= 4294967296) return 0xffffffff
  return u32.!(f);
}
```
Arguably, the first if condition is not required and we could do with removing it.

Reply:
I meant, compare the result of the conversion using an unsigned 64-bit integer comparison.

Reply:
Also, this function needs some documentation, ~2 sentences about what the inputs/outputs are and what it does.